File: Problem3_euler.py
num = 600851475143

def isprime(x):
    for i in range (2, x//2):
        if x % i == 0:
            return False
    return True

while not isprime(num):
    for i in range(3,num):
        if isprime(i):
            if num % i == 0:
                num = num // i
                break


print(num)

File: Problem3_euler.R
num <- 600851475143

is_prime <- function(x) {
  ceiling = x %/% 2
  for (i in 2:ceiling) {
    if (x %% i == 0) {
      return(FALSE)
    }
  }
  return(TRUE)
}


while (!is_prime(num)) {
  for (i in 3:num) {
    if (is_prime(i)) {
      if (num %% i == 0) {
        num <- num / i
        break
      }
    }
  }
}

print(num)